### PR TITLE
Updates to devcontainer.json to follow the latest dev container spec

### PR DESCRIPTION
**Pull request by krishnakumarg1984**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,27 +1,25 @@
...
    * Open VS Code and add the `Remote - Containers` extension `ms-vscode-remote.remote-containers`
    * Click on the green remote containers status button at the bottom left of the window
    * Then to check everything works select the `Remote-Containers: Try a Sample...` and the C++ example
    * Go through the README.md and suggested things to try to familiarise yourself  
  
-To use the Ubuntu 20.04 dev container you should clone this repository to your host machine and then copy the `.devcontainer` folder including the devcontainer.json and Dockerfile files inside it to the cmake project you want to develop:
-`cp -r .devcontainer /path/to/YourCMakeProject/`.
-
-You can then open `YourCMakeProject` inside VS Code and it will ask if you want to reopen and build the project in the dev container that we defined. Once you have done that you can build, compile, debug and use the VS Code terminal from within the dev container. Try opening a VS Code terminal and typing `cat /etc/os-release` and you should see the the OS is Ubuntu 20.04.1 LTS (Focal Fossa). 
+To use this Ubuntu 20.04 dev container, you should configure this repository as a submodule of your main git repository.
```
Comment (JamieJQuinn) [inline README.md]:
```suggestion
To use this Ubuntu 20.04 dev container, you should configure this repository as a submodule of your main git repository or download the `.devcontainer` folder and include it in the root directory of your project.
```

Comment (JamieJQuinn) [reply README.md]: A student might (fairly) want to use the devcontainer without using submodules.

**Review by JamieJQuinn — CHANGES_REQUESTED** [review]
Just one suggestion, otherwise *wonderful* work modernising this!

File: README.md (hunk 1 above)
Comment (krishnakumarg1984) [reply README.md]: Agreed, and accepted the suggested edit.

**Review by JamieJQuinn — APPROVED** [review]
Beautiful!

**Pull request merged**